docs/articles/basic-usage.md

```diff
@@ -0,0 +1,221 @@
+  "pet": {"$ref": "#/definitions/Pet"}
+}
+```
+
+In Bonsai, they can be manipulated as [`Enum`](https://learn.microsoft.com/en-us/dotnet/api/system.enum?view=net-9.0) types:
+
+:::workflow
+![Person and Pets](~/workflows/person-and-pet-enum.bonsai)
+:::
+
+> [!TIP]
+> In certain cases, it may be useful to use `x-enum-names` to specify the rendered names of the enum values.
+>
+> ```json
+> {
+>   "MyIntEnum": {
+>     "enum": [0, 1, 2, 3, 4],
+>     "title": "MyIntEnum",
+>     "type": "integer",
+>     "x-enumNames": ["None", "One", "Two", "Three", "Four"]
+>   }
+> }
+> ```
+
+## Lists
+
+`Bonsai.Sgen` also supports the generation of lists using the `array` type in the `json-schema`:
+
+```json
+"pets": {
+  "type": "array",
+  "items": {"$ref": "#/definitions/Pet"}
+}
+```
+
+`json-schema` `array` will be rendered as [`List<T>`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1?view=net-9.0) in the generated code and can be manipulated (and created) as such.
+
+:::workflow
+![Person and Pets](~/workflows/person-and-pets-enum.bonsai)
+:::
+
+## Nullable types
+
+`json-schema` supports the `null` type, which can be used to represent nullable types. The standard is a bit loose in this regard, but `Bonsai.Sgen` will generate a nullable-T if the json-schema represents it using the `oneOf` keyword:
+
+```json
+"pet": {
+  "oneOf": [
+    {"$ref": "#/definitions/Pet"},
+    {"type": "null"}
+  ]
+}
+```
+
+For value types, the generated code will render a [Nullable value type](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types) type. This type will expose two properties: `HasValue` and `Value`, that can be used to test and manipulate the type, respectively.
+
+For reference types, the generated code will not render a nullable type since reference types are already nullable in C#. An application can test for `null` to determine if the value is present by simply using an `ExpressionTransform` operator with `it == null`:
+
+:::workflow
+![Nullable pet](~/workflows/person-and-pet-enum-nullable.bonsai)
+:::
+
+## Required fields
+
+`json-schema` supports the [`required`](https://json-schema.org/learn/getting-started-step-by-step#define-required-properties) keyword to specify which fields are required. By default, all fields are optional. This can be useful to enforce the presence of certain fields in the object at deserialization time. However, `Bonsai.Sgen` will not generate any code to enforce this requirement during object construction, only at deserialization. It is up to the user to ensure that the object is correctly populated before using it.
+
+> [!Note]
+> Some confusion may arise about the distinction between `null` and `required`. This is all the more confusing since different languages and libraries may refer to these concepts differently. For the sake of this tool (and honestly in general) the following definitions are used:
```

Comment:
```suggestion
> Some confusion may arise about the distinction between `null` and `required`. This is all the more confusing since different languages and libraries may refer to these concepts in different ways. For the sake of this tool, the following definitions are used:
```